examples/tutorial/basic-5/programs/basic-5/src/lib.rs

```diff
@@ -0,0 +1,109 @@
+use anchor_lang::prelude::*;
+
+declare_id!("DuT6R8tQGYa8ACYXyudFJtxDppSALLcmK39b7918jeSC");
+
+#[program]
+pub mod basic_5 {
+    use super::*;
+
+    pub fn create(ctx: Context<Create>) -> Result<()> {
+        let action_state = &mut ctx.accounts.action_state;
+        // * - means dereferencing
+        action_state.user = *ctx.accounts.user.key;
+        // Lets initialize the state
+        action_state.action = 0;
+
+        Ok(())
+    }
+    
+    pub fn walk(ctx: Context<Walk>) -> Result<()> {
+        let action_state = &mut ctx.accounts.action_state;
+        // Lets change the robot action state to "walk"
+        action_state.action = 1;
+
+        Ok(())
+    }
+    
+    pub fn run(ctx: Context<Run>) -> Result<()> {
+        let action_state = &mut ctx.accounts.action_state;
+        // Lets change the robot action state to "run"
+        action_state.action = 2;
+
+        Ok(())
+    }
+    
+    pub fn jump(ctx: Context<Jump>) -> Result<()> {
+        let action_state = &mut ctx.accounts.action_state;
+        // Lets change the robot action state to "jump"
+        action_state.action = 3;
+        
+        Ok(())
+    }
+
+    pub fn reset(ctx: Context<Reset>) -> Result<()> {
+        let action_state = &mut ctx.accounts.action_state;
+        // Lets reset the robot action states
+        action_state.action = 0;
+        
+        Ok(())
+    }
+}
+
+#[derive(Accounts)]
+pub struct Create<'info> {
+    // init means to create action_state account
+    // bump to use unique address for action_state account
+    #[account(init, payer=user, space= 8 + ActionState::INIT_SPACE, seeds=[b"action-state".as_ref(), user.key().as_ref()], bump)]
```

Comment:
This line is very long, let's use multiple lines.

```suggestion
    #[account(
        init,
        payer = user,
        space= 8 + ActionState::INIT_SPACE,
        seeds = [b"action-state", user.key().as_ref()],
        bump
    )]
```

Reply:
Thanks for this correction, this has done.